src/pages/office/components/fault-comment/fault-comment.ts

```diff
@@ -20,14 +24,15 @@ export class FaultCommentComponent implements OnChanges {
   @Output()
   faultCommentChange = new EventEmitter<CommentedCompetency>();
 
-  private formControl: FormControl;
+  constructor(private outcomeBehaviourProvider: OutcomeBehaviourMapProvider) { }
 
   ngOnChanges(): void {
-    if (!this.formControl) {
-      this.formControl = new FormControl(null, Validators.required);
-      this.parentForm.addControl(this.formControlName, this.formControl);
+    if (this.outcomeBehaviourProvider.getVisibilityType(this.outcome, 'faultComment') === 'N') {
```

Comment:
magic string.
what is `N` by the way?

Reply:
introduced an enum so that intention is clearer (N is essentially Not visible)